src/overlay/test/OverlayTests.cpp

```diff
@@ -2829,31 +2885,62 @@ TEST_CASE("overlay pull mode loadgen", "[overlay][pullmode][acceptance]")
         3 * Herder::EXP_LEDGER_TIMESPAN_SECONDS, false);
 
     auto& loadGen = node1->getLoadGenerator();
+    if (txSizeLimit > 0)
+    {
+        node1->getHerder().setMaxTxSize(txSizeLimit);
+        node2->getHerder().setMaxTxSize(txSizeLimit);
+    }
 
     // Create 5 txns each creating one new account.
     // Set a really high tx rate so we create the txns right away.
     loadGen.generateLoad(GeneratedLoadConfig::createAccountsLoad(
         /* nAccounts */ numAccounts * MAX_OPS_PER_TX,
-        /* txRate */ 1));
+        /* txRate */ 2));
 
     // Let the network close multiple ledgers.
     // If the logic to advertise or demand incorrectly sends more than
     // they're supposed to (e.g., advertise the same txn twice),
     // then it'll likely happen within a few ledgers.
-    simulation->crankUntil(
-        [&] { return simulation->haveAllExternalized(5, 1); },
-        10 * Herder::EXP_LEDGER_TIMESPAN_SECONDS, false);
+    auto crank = [&]() {
+        simulation->crankUntil(
+            [&] { return simulation->haveAllExternalized(5, 1); },
+            10 * Herder::EXP_LEDGER_TIMESPAN_SECONDS, false);
+    };
 
-    // Node 1 advertised 5 txn hashes to each of Node 2 and Node 3.
-    REQUIRE(getAdvertisedHashCount(node1) == numAccounts);
-    REQUIRE(getAdvertisedHashCount(node2) == 0);
+    if (txSizeLimit == INVALID_LIMIT)
+    {
+        REQUIRE_THROWS_AS(crank(), std::runtime_error);
+    }
+    else
+    {
+        crank();
+        // Node 1 advertised 5 txn hashes to each of Node 2 and Node 3.
+        REQUIRE(getAdvertisedHashCount(node1) == numAccounts);
+        REQUIRE(getAdvertisedHashCount(node2) == 0);
+
+        REQUIRE(overlaytestutils::getSentDemandCount(node2) > 0);
+        REQUIRE(overlaytestutils::getFulfilledDemandCount(node1) ==
+                numAccounts);
 
-    REQUIRE(overlaytestutils::getSentDemandCount(node2) > 0);
-    REQUIRE(overlaytestutils::getFulfilledDemandCount(node1) == numAccounts);
+        // As this is a "happy path", there should be no unknown demands.
+        REQUIRE(getUnknownDemandCount(node1) == 0);
+        REQUIRE(getUnknownDemandCount(node2) == 0);
 
-    // As this is a "happy path", there should be no unknown demands.
-    REQUIRE(getUnknownDemandCount(node1) == 0);
-    REQUIRE(getUnknownDemandCount(node2) == 0);
+        if (configs[0].EXPERIMENTAL_TX_BATCH_MAX_SIZE > 0)
+        {
+            auto& om = node1->getOverlayManager().getOverlayMetrics();
+            if (txSizeLimit > 0)
```

Comment:
I'm not sure if I follow this if statement, what are we checking here?

Reply:
there's a comment on line 2837 wrt txSizeLimit - if it's 0, we don't enforce size limits, so we'll get maximum size batches. Otherwise, our batch size depends on the tx size limit.